playground/backend/internal/code_processing/code_processing.go

```diff
@@ -65,7 +68,7 @@ func Process(ctx context.Context, cacheService cache.Cache, lc *fs_tool.LifeCycl
 	validateFunc := executor.Validate()
 	go validateFunc(successChannel, errorChannel)
 
-	if err := processStep(ctxWithTimeout, pipelineId, cacheService, cancelChannel, successChannel, nil, nil, errorChannel, pb.Status_STATUS_VALIDATION_ERROR, pb.Status_STATUS_PREPARING); err != nil {
+	if err := processStep(ctxWithTimeout, pipelineId, cacheService, cancelChannel, successChannel, nil, nil, nil, nil, errorChannel, pb.Status_STATUS_VALIDATION_ERROR, pb.Status_STATUS_PREPARING); err != nil {
```

Comment:
I agree with @ilya-kozyrev about the number of arguments – it's quite big. One possible solution I see is to create a class `CodeProcessor` which will store all these arguments that are reused everywhere. I.e we can store all of the following args in the class: `ctxWithTimeout, pipelineId, cacheService, cancelChannel, successChannel, errorChannel`

Reply:
Changed.